nibabel/funcs.py

```diff
@@ -224,7 +224,18 @@ def as_closest_canonical(img, enforce_diag=False):
     # we need to transform the data
     arr = img.get_data()
     t_arr = apply_orientation(arr, ornt)
-    return img.__class__(t_arr, out_aff, img.header)
+    # Also apply the transform to the dim_info fields
+    new_hdr = img.header.copy()
+    new_dim = list(new_hdr.get_dim_info())
+    for idx, value in enumerate(new_dim):
```

Comment:
Hi Matthew,

We don't permute the None's because this array is (freq, phase, slice), *not* the order of the actual data axes.

As an example, if we have a "normal" MRI image in which
freq = 0, phase = 1, slice = 2
and we permute dimensions 1 and 2, we obviously end up with
freq = 0, phase = 2, slice = 1

If we have an image in which one or more of freq, phase and slice are None (e.g. spiral images):
freq = None, phase = None, slice = 1
if we then permute axes 1 and 2, we move the slice dimension to dimension 2, but we still don't have any mapping for the freq and phase attributes:
freq = None, phase = None, slice = 2

I don't know if I'm doing a good job of explaining this...